src/main/src/main/java/org/geoserver/catalog/FeatureTypeInfo.java

```diff
@@ -193,4 +193,13 @@
 	
 	void setCircularArcPresent(boolean arcsPresent);
 
+
+	/**
+	 * Whether querying the full number of features is banned for this feature type.
+	 * 
+	 * @return True if a full feature count should not be attempted, false otherwise
+	 */
+	boolean isTotalCountBanned();
```

Comment:
The primitive defaults to false which means business as usual. Updating your configuration with a new geoserver then ends up writing this value to the config and thus upgrading the configuration "naturally". I think this is the way to go.